helion/_testing.py

```diff
@@ -30,6 +34,224 @@
+
+        # Patch self.assertRaises to count calls
+        if RefEagerTestBase._original_assert_raises_func is None:
+            RefEagerTestBase._original_assert_raises_func = self.assertRaises
+
+        def counting_assert_raises(*args: object, **kwargs: object) -> object:
+            RefEagerTestBase._assert_raises_count += 1
+            return RefEagerTestBase._original_assert_raises_func(*args, **kwargs)  # type: ignore[misc]
+
+        self.assertRaises = counting_assert_raises
+
+        # Patch self.skipTest to count calls
+        if RefEagerTestBase._original_skip_test_func is None:
+            RefEagerTestBase._original_skip_test_func = self.skipTest
+
+        def counting_skip_test(*args: object, **kwargs: object) -> object:
+            RefEagerTestBase._skip_test_count += 1
+            return RefEagerTestBase._original_skip_test_func(*args, **kwargs)  # type: ignore[misc]
+
+        self.skipTest = counting_skip_test
+
+        # Store the tracking context manager instance so we can check counts in tearDown
+        self._run_ref_tracker = track_run_ref_calls()
+        self._run_ref_count = self._run_ref_tracker.__enter__()
+
+    def tearDown(self) -> None:
+        """Common teardown with assertion counting check."""
+        # If not in ref eager mode, skip the teardown logic
+        if not self._in_ref_eager_mode:
+            super().tearDown()  # type: ignore[misc]
+            return
+
+        try:
+            # Exit the run_ref tracker
+            self._run_ref_tracker.__exit__(None, None, None)
+
+            # Check if the test was skipped
+            test_method = getattr(self, self._testMethodName, None)  # type: ignore[attr-defined]
+            is_skipped = (
+                test_method is not None
+                and hasattr(test_method, "__unittest_skip__")
+                and test_method.__unittest_skip__
+            ) or RefEagerTestBase._skip_test_count > 0
+
+            # Assert that either run_ref was called or the test was skipped
+            if not is_skipped and self._run_ref_count[0] == 0:
+                self.fail(  # type: ignore[attr-defined]
+                    f"Test {self._testMethodName} did not call run_ref and was not skipped"  # pyright: ignore[reportAttributeAccessIssue]
+                )
+
+            if not is_skipped:
+                # Check that either assert_close, assertRaises, or skipTest was called
+                total_assertions = (
+                    RefEagerTestBase._assert_close_count
+                    + RefEagerTestBase._assert_raises_count
+                    + RefEagerTestBase._skip_test_count
+                )
+                self.assertGreater(  # type: ignore[attr-defined]
+                    total_assertions,
+                    0,
+                    f"Test {self._testMethodName} did not call torch.testing.assert_close, assertRaises, or skipTest",  # type: ignore[attr-defined]  # pyright: ignore[reportAttributeAccessIssue]
```

Comment:
> even for the tests that are enabled we are pretty aggressive disabling stuff like assertTrue and assertEqual

@jansel we have a requirement here that all enabled tests must either call `torch.testing.assert_close` or `assertRaises` or `self.skipTest`, so I believe even though assertTrue and assertEqual etc. are disabled, we are still checking correctness via those enabled assertions